pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1557,7 +2238,7 @@ func modelToEntry(tx *gorm.DB, model RegisteredEntry) (*common.RegistrationEntry
 	}
 
 	var fetchedDNSs []*DNSName
-	if err := tx.Model(&model).Related(&fetchedDNSs).Error; err != nil {
+	if err := tx.Model(&model).Related(&fetchedDNSs).Order("registered_entry_id ASC").Error; err != nil {
```

Comment:
Is the sort required?

Reply:
Yes, this sort is required. There are semantics around the  order of DNS names in the registration entry so the ordering must be preserved by the datastore ( first name also becomes the CommonName of the certificate). We don't have an explicit column to use to order these in the database. With the current model, the id should be a good indicator of insertion order (barring id re-use), which is better than what we currently have.

Reply:
ok, we should take a stab at documenting this in the proto if it isn't already. Can be a follow up, I see why its required here but I wouldn't want other implementations to have to learn this lesson.